tpm2_pytss/tsskey.py

```diff
@@ -0,0 +1,20 @@
+"""
+SPDX-License-Identifier: BSD-3
+"""
+
+
+class TSSKey(object):
+    def __init__(self, private, public, emptyAuth=True, parent=lib.TPM2_RH_OWNER):
```

Comment:
No public, public's not needed and uses the existing TPM2B_PUBLIC and PEM formats. `password=None`, and then detect in the code if password is emptyAuth or not.

Reply:
But the format contains the public key, so it's needed for serialization and will be read during deserialization.

I rather see password at load time, so it can be set directly on the ESYS_TR handle, I suspect that those who would use the class directly (not using fromPEM) will do so to call toPEM directly afterwards

Reply:
> But the format contains the public key, so it's needed for serialization and will be read during deserialization.
> 

Yeah my mistake. I realized this moments after I commented. "ts the TPM private and public blobs from tpm2_load calls.

> I rather see password at load time, so it can be set directly on the ESYS_TR handle, I suspect that those who would use the class directly (not using fromPEM) will do so to call toPEM directly afterwards

OK so the flow (psuedo code) would be:
```Python
pub, priv = Esys_Create(someParent, Template)
key = TSSKey(pub, priv, emptyAuth=True, parent=SomeParent)
pem = key.toPEM()
file.write(pem)
```
or:

```python
pem = file.read()
key = TSSKey.fromPEM(pem)
ctx = Esys_Load(key.pub, key.priv)
# use key
```

I like it. We're just going to want some @property gettr's for getting at the various fields.

Reply:
I'll add the property methods.

I was thinking of having load and create methods attached to the class, and the reason for that is how tpm2-openssl handles the parent, it's either by template (if parent is TPM2_RH_OWNER), but it might be a persistent key.

Reply:
FYI When I started taking over the Python bindings, the original design had the Esys_TR_setAuth call on the ESYS_TR object itself. So the ESYS_TR object would also have a handle to ectx. Which means you have to be careful that the ESYS context pointer is not freed and closed out from underneath you. So we want to make sure we don't internally keep any esys ctx, just make the caller always pass it. That way the burden of ectx lifecycle is on them.

FYI I also didn't tie it to __del__ because we can't control when or if it's called. In the testing when I had tried this, it would hang occasionally as the ectx wouldn't get closed and the other one would wait while trying to establish a connection with the TPM. Thus you couldn't write code like:

```python
ectx = Esys_Context()
ectx.doSomething()

# This hangs because ectx __del__ was'nt always called
ectx2 = Esys_Context()
ectx2.doSomething()
```